### ISSUE TITLE
Move Stepper Input to core

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#4019]
Follow on from #1855

**Comment by mark-tate** [comment]
Lungo Goal: push to core if Jake happy

**Comment by jake-costa** [comment]
Found a few items to address. Will add feedback to teams channel by EOD.

**Comment by jake-costa** [comment]
Feedback added to teams channel

**Comment by origami-z** [comment]
> URL: https://storybook.saltdesignsystem.com/?path=/story/lab-stepper-input--default
>  
> When dev picks this up let me know and we can discuss these to go over any questions and possible remediation suggestions.
>  
> Bordered Example (Usability not accessibility):
> When the component receives focus initially the value does not increase/decrease when using arrow keys or when selecting the up/down buttons using a mouse. These are initially disabled.
> After a number is entered the value does increase/decrease when using arrow keys and when selecting the up/down buttons using a mouse.
>  
> Read-only Example:
> Does not convey it is read-only status to screen reader users. It says “Stepper Input Spin button editable 5”
>  
> Validation Example:
> Does not convey the validation state to screen reader users. Please note: I did modify the visible label as that was the only way the status was being conveyed but is not going to accurately reflect the experience when implemented.
>  
> Refresh Adornment & Controlled Example:
> Screen readers do not covey anything when the refresh button is pressed
>  
> Custom Buttons Example:
> The plus and minus buttons are in the focus order. These should be removed similar to the up/down arrow buttons in the other examples.

**Comment by mark-tate** [comment]
Macc Goal: no work planned, capacity issues

**Comment by bhoppers2008** [comment]
Awaiting feedback for dev.

**Comment by mark-tate** [comment]
Paprika Goal: turn into a known quantity by EOS
We an make our own decision on merging this to Core, KISS.

**Comment by mark-tate** [comment]
Paprika: Review by EOS, draft PR raised

**Comment by pseys** [comment]
Dependent on #4116 and #4798

**Comment by mark-tate** [comment]
Paprika: thickness and gap tokens required by EOS, 

Separate PR
@tayyabataimur @ivan-calderon  to help get this into PR by EOS

**Comment by tayyabataimur** [comment]
- Draft PR raised, actioning A11y feedback sent over by @jake-costa.
- Have a kickoff call scheduled on 13 March to discuss gap and thickness tokens strategy with @ivan-calderon and @pseys

**Comment by mark-tate** [comment]
Cayenne Goal: a11y feedback, will be NumberInput in code and design will follow later
Consider other doc changes which may already reference StepperInput.
Merged to Labs by EOS

**Comment by tayyabataimur** [comment]
PR open for renaming StepperInput to NumberInput https://github.com/jpmorganchase/salt-ds/pull/4850

**Comment by tayyabataimur** [comment]
PR open for fixing A11y issues: https://github.com/jpmorganchase/salt-ds/pull/4829

**Comment by mark-tate** [comment]
Cardamom Goal: Agreement on what we are releasing and any new features published in Labs by EOS

**Comment by mark-tate** [comment]
Dill Goal: Agree/Align spec to code and release to Labs.
Aim for Clove release, if it's in a known state

**Comment by origami-z** [comment]
Kickoff scheduled

**Comment by origami-z** [comment]
May 1 - extract format logic in code starting soon, update with WCAG 2.2 design proposal to BLM

**Comment by mark-tate** [comment]
draft pocs to be shared to get feedback

**Comment by mark-tate** [comment]
Allspice goal: Identify a sprint when this will ship to Core

- how do we get it from Labs to Core ?
- extracted API to be shared via draft PR
- design decision/review

**Comment by joshwooding** [comment]
Working on addressing feedback, will push for a review after?
